lib/pry/config/behavior.rb

```diff
@@ -231,7 +231,7 @@ def __dup(value)
   def __push(key, value)
     define_singleton_method(key) { self[key] } if not singleton_class.method_defined? key
     define_singleton_method("#{key}=") {|value| @lookup[key] = value } if not singleton_class.method_defined? "#{key}="
-    send("#{key}=", value)
+    @lookup[key] = value
```

Comment:
@dgutov Can you review this if you have time, please? 
Using `send()` here introduced a case of infinite recursion. I solved in this commit, via a different way.

Reply:
I kind of liked the idea behind that code: define the methods if they don't exist yet, and call them afterwards. Maybe you could deal with the infloop if, instead of redefining the method on the instance (this is what you're doing, right?), you create an anonymous module and prepend it, and define those methods there.

But if the alternative approach works all right, guess it's fine too.

Reply:
@dgutov Sorry. I didn't see your reply earlier. Yeah, maybe we could change it back later but it's not as high priority for me as getting this branch in a working state. I don't have a strong feelings one way or the other, except it seems flawed:

```ruby
define_singleton_method("#{key}=") {|value| @lookup[key] = value } if not singleton_class.method_defined? "#{key}="
send("#{key}=", value)
```

What if we didn't define the singleton method? Can we trust the implementation to be `@lookup[key]=...`?

Reply:
> What if we didn't define the singleton method? Can we trust the implementation to be @lookup[key]=...?

Now that Readline setup is performed somewhere else? I guess we can.

Reply:
With this PR, `#{key}=` can (re)defined. It seems to have been the cause, where Pry::Deprecate defines it first, so our `@lookup[key] = ...` implementation is not registered. But i'm not sure.

Reply:
That might be the case.

One way to get around this would be to replace some of the snazzy metaprogramming with delegation. That is, have a Config::Deprecations object wrap the Config object, instead of having all methods defined, and redefined, on one object.